fix/gifted-pets.lua

```diff
@@ -0,0 +1,112 @@
+-- Fixes pets gifted to another unit in Adventure Mode
+-- Fixes pets owned by companions missing from the companions list
+
+local function isPetOf(petNemesis, ownerNemesis)
+    if petNemesis.unit then
+        local owner_id = petNemesis.unit.relationship_ids.Pet
+        if owner_id ~= -1 and owner_id == ownerNemesis.unit_id then
+            return true
+        end
+    elseif petNemesis.figure then -- in case the unit is offloaded
+        for _, link in ipairs(petNemesis.figure.histfig_links) do
+            if link._type == df.histfig_hf_link_pet_ownerst and ownerNemesis.figure and link.target_hf == ownerNemesis.figure.id then
+                return true
+            end
+        end
+    end
+    return false
+end
+
+
+local function findPetOwnerOf(petNemesis)
+    if petNemesis.figure then
+        for _, link in ipairs(petNemesis.figure.histfig_links) do
+            if link._type == df.histfig_hf_link_pet_ownerst then
+                local target_hf = link.target_hf
+                return df.nemesis_record.find(df.historical_figure.find(target_hf).nemesis_id)
+            end
+        end
+    end
+    return nil
+end
+
+local function findInParty(party, histfig_id)
+    for k, v in ipairs(party) do
+        if v == histfig_id then
+            return true
+        end
+    end
+    return false
+end
```

Comment:
this could be replaced with a call to `utils.linear_index` (or `utils.binsearch` if the vector is sorted)